### PR TITLE
chore: remove deprecated functions url and ugettext_lazy

### PR DESCRIPTION
**Pull request by gagantrivedi**
Thanks for submitting a PR! Please check the boxes below:

- [x] I have run [`pre-commit`](https://docs.flagsmith.com/platform/contributing#pre-commit) to check linting
- [ ] I have added information to `docs/` if required so people know about the feature!
- [x] I have filled in the "Changes" section below?
- [x] I have filled in the "How did you test this code" section below?
- [x] I have used a [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) title for this Pull Request

## Changes

- Replace `django.conf.urls url` with `django.urls.re_path`
- Replace `ugettext_lazy` with `gettext_lazy`

Note: ugettext_lazy and url were deprecated in Django 3.x.

## How did you test this code?

covered by existing unit tests( :crossed_fingers: )

### CODE REVIEW

File: api/app/urls.py [hunk 1]
```diff
@@ -70,4 +69,4 @@
 
 if settings.SERVE_FE_ASSETS:
```
Comment (matthewelwell) [inline api/app/urls.py]:
```suggestion
if settings.SERVE_FE_ASSETS:  # pragma: no cover
```

File: api/environments/models.py [hunk 2]
```diff
@@ -10,7 +10,7 @@
 from django.db import models
 from django.db.models import Prefetch, Q
 from django.utils import timezone
-from django.utils.translation import ugettext_lazy as _
+from django.utils.translation import gettext_lazy as _
```
Comment (matthewelwell) [inline api/environments/models.py]: I had a look at where we're actually using this and I think we can just remove it, right?

File: api/features/models.py [hunk 3]
```diff
@@ -21,7 +21,7 @@
 from django.db import models
 from django.db.models import Max, Q, QuerySet
 from django.utils import timezone
-from django.utils.translation import ugettext_lazy as _
+from django.utils.translation import gettext_lazy as _
```
Comment (matthewelwell) [inline api/features/models.py]: Probably the same comment as above.

File: api/users/urls.py [hunk 4]
```diff
@@ -24,4 +23,4 @@
 ]
 
 if settings.ALLOW_ADMIN_INITIATION_VIA_URL:
```
Comment (matthewelwell) [inline api/users/urls.py]:
```suggestion
if settings.ALLOW_ADMIN_INITIATION_VIA_URL:  # pragme: no cover
```

File: api/environments/models.py (hunk 2 above)
Comment (gagantrivedi) [reply api/environments/models.py]: Yeah, we don't really support any other language; I will remove them

Comment (gagantrivedi) [reply api/environments/models.py]: Done

File: api/features/models.py (hunk 3 above)
Comment (gagantrivedi) [reply api/features/models.py]: Done

**Review by matthewelwell — APPROVED** [review]

**Pull request merged**